lib/engine/route.rb

```diff
@@ -285,6 +299,13 @@ def connection_data
       @connection_data = []
       return @connection_data unless @connection_hexes
 
+      if @train.local? && @connection_hexes.one? && @connection_hexes[0].include?('local')
```

Comment:
why do you need to check include? local what does that do

Reply:
I explaned the whole reason for the 'local' in the comment above. But realised it might be a wall of text :)

To get the last_route to work i first modify the connection_hexes method a bit. Since there are no path in the connection_data object no data was returned from the id method on in the connection object. Here i opted to save the local connection info with a format of ['local', hex.id]. The action entry now looks the same for the local as it does for any normal path object, with the exception for the 'local'.

The 'local' is need in the connection_data method when we play back the actions, here i check if the train is local? and the connected_hex array includes a 'local'. If so create the same connection object as what i did in touch_node method. With this previous route now also works.